extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-manager.ts

```diff
@@ -222,6 +226,17 @@ export class ChePluginManager {
       registries[registry.name] = registry;
     }
 
+    const devfile = await this.devfileService.get();
+    if (devfile.components) {
+      const components = devfile.components.filter(component => component.plugin && component.plugin.id && component.plugin.registryUrl);
+      for (let i = 0; i < components.length; i++) {
+        const component = components[i];
```

Comment:
can be simplified like: 
```
for (const component of components) {
```
but up to you

Reply:
done